### DISCUSSION TITLE
Equities.search() by index (e.g. S&P500)

### DISCUSSION THREAD

**Discussion by cianryan09** [JerBouma/FinanceDatabase#69] [Ideas]
I think this would be a great enhancement if not already available. An obvious example is the S&P500 where the constituents are across two exchanges (NYSAE/NASDAQ). You can only get so far by filtering by exchange then market cap size. 
the ability to filter down for other larger and less common indices such as the FTSE Russell Global index series (GEIS) where they have 10k+ constituents would also be really useful.

**Comment by JerBouma** [comment]
Hi @cianryan09, I think that is an excellent idea. The only major issue will be that companies drop out of the index over time so there needs to be some system in place that takes into account these changes. Do you have any suggestions? Otherwise it becomes pretty difficult to keep it up to date. E.g: https://en.wikipedia.org/wiki/List_of_S%26P_500_companies#Selected_changes_to_the_list_of_S&P_500_components

**Reply by cianryan09** [reply]
Hi, happy new year and please forgive the late response! I'm glad you find the idea worthwhile. Yes, since the index changes it will need to look up a 'live' version of the index from a database when called. For looking up the S&P, I am using the following code:

```
import pandas as pd

def list_spy_holdings() -> pd.DataFrame:
    # Ref: https://stackoverflow.com/a/75845569/
    # Source: https://www.ssga.com/us/en/intermediary/etfs/funds/spdr-sp-500-etf-trust-spy
    # Note: One of the included holdings is CASH_USD.
    url = 'https://www.ssga.com/us/en/intermediary/etfs/library-content/products/fund-data/etfs/us/holdings-daily-us-en-spy.xlsx'
    return pd.read_excel(url, engine='openpyxl', index_col='Ticker', skiprows=4).dropna()

SPY_df = list_spy_holdings()

ticker_list = SPY_df.index.tolist()
```
This uses the constituents of the SPDR ETF rather than the index itself, although they should be the same anyway.

I know it's a different library, but the Toolkit has a benchmark variable and the S&P as the default. Is it already looking this up in some way (unless it's just using the ETF ticker)? It can also be set to the Dow Jones Average (DJIA) if I'm not mistaken - maybe some indices are already on the underlying databases (financial modeling prep and Yahoo)?
